packages/http-server-javascript/package.json

```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@typespec/http-server-javascript",
+  "version": "0.55.0",
+  "author": "Microsoft Corporation",
+  "description": "TypeSpec HTTP server code generator for JavaScript",
+  "homepage": "https://github.com/microsoft/typespec",
+  "readme": "https://github.com/microsoft/typespec/blob/main/packages/http-server-javascript/README.md",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/microsoft/typespec.git"
+  },
+  "bugs": {
+    "url": "https://github.com/microsoft/typespec/issues"
+  },
+  "keywords": [
+    "typespec",
+    "http",
+    "server",
+    "javascript",
+    "typescript"
+  ],
+  "type": "module",
+  "main": "dist/src/index.js",
+  "tspMain": "lib/main.tsp",
+  "scripts": {
+    "clean": "rimraf ./dist ./temp",
+    "build": "npm run build:helpers && npm run build:src",
+    "build:src": "tsc -p ./tsconfig.json",
+    "build:helpers": "tsc -p ./tsconfig.helpers.json && tsx ./build-helpers.ts",
+    "watch": "tsc -p . --watch",
+    "test": "echo No tests specified",
+    "test:ci": "echo No tests specified",
+    "lint": "eslint . --max-warnings=0",
+    "lint:fix": "eslint . --fix",
+    "regen-docs": "tspd doc .  --enable-experimental  --output-dir ../../docs/libraries/http-server-javascript/reference"
+  },
+  "devDependencies": {
+    "@types/node": "~18.11.19",
+    "@typespec/compiler": "workspace:~",
+    "@typespec/http": "workspace:~",
+    "typescript": "~5.5.3",
+    "prettier": "~3.3.2",
```

Comment:
should prettier be a dev dependency?

Reply:
Well, from what I've seen we consider all the dependencies of the emitters to be dev dependencies. Prettier is actually invoked in the emitter, so I could put it in dependencies or devDependencies either way. The protobuf emitter for example has no actual dependencies. Just devDependencies.

Reply:
ah if you are calling it it should be a dependency, also notice now but the compiler and http should be peerDependencies.

Rule is:
- Typespec deps should be `peerDependencies`(and optionall devDependencies too to have a more specific range)
- Other package deps that are used at runtime `dependencies`
- PAckages just used in test/build `devDependencies`